### PR TITLE
Update prod builders.

### PR DESCRIPTION
**Pull request by godofredoc**
## Description

This is to make the builders show up in the flutter dashboard.

## Related Issues

Bug:
  https://github.com/flutter/flutter/issues/63256

## Tests

N/A

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by keyonghan — COMMENTED** [review]
Are we ready to make these changes?
If updated
- These builders will show up on dashboard, together with old `Linux`, `Windows`, and `Mac`. New ones will have limited records on the top, whereas old ones with be empty for recent runs. If this is to be expected, shall we inform the team first?
- Issue https://github.com/flutter/flutter/issues/63256 needs to be addressed first before updating the builders here. Otherwise, push_build_status_to_github API will give the incorrect result.

One question:
where is the point to enable/kick-off these prod builders whenever a PR is submitted? Cocoon seems to only refresh status of existing running/completed prod builder.

**Comment by godofredoc** [comment]
> Are we ready to make these changes?
> If updated
> 
> * These builders will show up on dashboard, together with old `Linux`, `Windows`, and `Mac`. New ones will have limited records on the top, whereas old ones with be empty for recent runs. If this is to be expected, shall we inform the team first?
I'll send an email when we have the PRs ready to get submitted.
> * Issue #63256 needs to be addressed first before updating the builders here. Otherwise, push_build_status_to_github API will give the incorrect result.
PR will be sent in a couple of mins.
> 
> One question:
> where is the point to enable/kick-off these prod builders whenever a PR is submitted? Cocoon seems to only refresh status of existing running/completed prod builder.
Not quite sure I follow your question. We only use this file to update the datastore with info from luci and to create the initial records. The one that needs to be tested on presubmit is try_builders.json

**Comment by keyonghan** [comment]
Re my last question. In other words, how does LUCI know which prod builders to trigger when a PR is submitted?

File: dev/prod_builders.json [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
+      },
+      {
+         "name": "Windows",
+         "repo": "flutter",
+         "taskName": "windows_bot",
+         "flaky": false
       }
    ]
-}
+}
```
Comment (keyonghan) [inline dev/prod_builders.json:46]: Nit: a new line.

**Review by keyonghan — APPROVED** [review]
LGTM.

**Comment by godofredoc** [comment]
> Re my last question. In other words, how does LUCI know which prod builders to trigger when a PR is submitted?

Those are configured in flutter/infra project and have been running for a couple of days. Once we enable the changes it will fill out to max limits configured in the cocoon backend.

https://ci.chromium.org/p/flutter/g/framework/console

File: dev/prod_builders.json (hunk 1 above)
Comment (godofredoc) [reply dev/prod_builders.json:46]: The formatter removes the line at the end.

**Comment by keyonghan** [comment]
you may notice that cocoon is using `taskName` for LuciBuilder: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/service/luci.dart#L175

**Comment by godofredoc** [comment]
> you may notice that cocoon is using `taskName` for LuciBuilder: https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/service/luci.dart#L175

This is actually a fix for that, the serialization is expecting a json task_name that will be translated to model property taskName

**Pull request merged**